### PR TITLE
Rc/location update warnings

### PR DESCRIPTION
**Pull request by Robert-Costello**
## Product Description

⚠️  Adds warnings when editing Web User and Mobile Worker locations. There are two scenarios where a warning will be shown:

1. When a user (Web or Mobile) is location restricted and is about to have all locations removed.
2. When the Web User making changes to another user (Web or Mobile) is about to remove the only location shared with the user being edited.

The warning text is specific to the scenario that triggered the warning.
Warnings are read aloud by screen reader if a screen reader is being used.

Screenshot 2024-11-15 at 3 27 24 PM

## Technical Summary

[Jira Ticket](https://dimagi.atlassian.net/browse/USH-4626)

When making changes to assigned location of location restricted users, the user making the changes can lose access to the user they are editing and the user being edited can lose access to CommCare altogether.  

### Safety story

Tested locally and on staging.

This only handles whether to show a warning depending on the selections made in the UI and does not make any changes to users/location data.

### QA Plan
no QA planned

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by Robert-Costello** [comment]
_"update page contexts for editing mobile and web users"_ should have been _"update initial page data"_ 
Leaving a comment in lieu of editing the commit.

File: corehq/apps/locations/static/locations/js/widgets.js [hunk 1]
```diff
@@ -145,6 +147,50 @@ hqDefine("locations/js/widgets", [
...
+            }
+        });
+        let shareLocations = false;
+        _.every(selectedLocations, function (locationId) {
+            if (_.contains(requestUserLocations, locationId)) {
+                shareLocations = true;
+            }
+        });
+
+        const noCommonLocationWarning = $(".no-common-location");
```
Comment (MartinRiese) [inline corehq/apps/locations/static/locations/js/widgets.js:176]: I think an id would be more appropriate than a class to identify a single element.

File: corehq/apps/locations/static/locations/js/widgets.js [hunk 2]
```diff
@@ -145,6 +147,50 @@ hqDefine("locations/js/widgets", [
...
+    function handleLocationWarnings(select) {
+        const requestDomain = initialPageData.get('domain');
+        const selectedLocations = select.val();
+        const requestUserDomainMemberships = initialPageData.get('request_user_domain_memberships');
+        const requestUserLocations = _.find(requestUserDomainMemberships, function (dm) {
+            if (dm.domain === requestDomain) {
+                return dm.assigned_location_ids;
+            }
+        });
+        let shareLocations = false;
```
Comment (MartinRiese) [inline corehq/apps/locations/static/locations/js/widgets.js:169]: I would have expected `can_access_all_locations` to be part of this calculation. To me it looks like there are two mechanisms that hide the warning `can_access_all_locations` for including the divs in the dom and this to hid the divs if they are present. Is that correct?

File: corehq/apps/locations/static/locations/js/widgets.js (hunk 1 above)
Comment (Robert-Costello) [reply corehq/apps/locations/static/locations/js/widgets.js]: agreed, changed here https://github.com/dimagi/commcare-hq/pull/35379/commits/fce39404d26a4dd6572e5ff93b961ec797c409f3

**Comment by Robert-Costello** [comment]
@MartinRiese pinging for final review

**Review by MartinRiese — APPROVED** [review]

**Pull request merged**